### PR TITLE
Convert MAIL_DEFAULT_SENDER to a string

### LINKED ISSUE #1005: MAIL_DEFAULT_SENDER format causes a crash on Docker
Hello,

So I'm hosting ihm with docker, and a mariadb container for the db. If I try to create a new project, I get a 500 error and the following error stack on the server. I can anyway access to this project if I try to log into it.

```
ihatemoney_1  | ERROR [ihatemoney.run] Exception on /create [POST]
ihatemoney_1  | Traceback (most recent call last):
ihatemoney_1  |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
ihatemoney_1  |     response = self.full_dispatch_request()
ihatemoney_1  |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1518, in full_dispatch_request
ihatemoney_1  |     rv = self.handle_user_exception(e)
ihatemoney_1  |   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 271, in error_router
ihatemoney_1  |     return original_handler(e)
ihatemoney_1  |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1516, in full_dispatch_request
ihatemoney_1  |     rv = self.dispatch_request()
ihatemoney_1  |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1502, in dispatch_request
ihatemoney_1  |     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
ihatemoney_1  |   File "/src/ihatemoney/web.py", line 92, in admin_auth
ihatemoney_1  |     return f(*args, **kws)
ihatemoney_1  |   File "/src/ihatemoney/web.py", line 333, in create_project
ihatemoney_1  |     flash_email_error(
ihatemoney_1  |   File "/src/ihatemoney/utils.py", line 56, in flash_email_error
ihatemoney_1  |     (admin_name, admin_email) = current_app.config.get("MAIL_DEFAULT_SENDER")
ihatemoney_1  | ValueError: too many values to unpack (expected 2)
```

### PR DESCRIPTION
**Pull request by Glandos**
Fixes #1005

### CODE REVIEW

File: ihatemoney/utils.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def flash_email_error(error_message, category="danger"):
     admin email as a contact if MAIL_DEFAULT_SENDER is set to not the
     default value and SHOW_ADMIN_EMAIL is True.
     """
-    (admin_name, admin_email) = current_app.config.get("MAIL_DEFAULT_SENDER")
+    admin_email = current_app.config.get("MAIL_DEFAULT_SENDER")
     error_extension = "."
     if admin_email != "admin@example.com" and current_app.config.get(
```
Comment (zorun) [inline ihatemoney/utils.py:58]: This also needs adaptation I think

Comment (Glandos) [reply ihatemoney/utils.py:61]: And you were right!
I now use the email address parser to be **sure** that we compare against the default value.

Comment (zorun) [reply ihatemoney/utils.py:61]: Looks good, thanks :)

**Comment by zorun** [comment]
Strange, the CI did not run...

File: ihatemoney/run.py [hunk 2]
```diff
@@ -102,6 +102,15 @@ def validate_configuration(app):
         if "MAIL_DEFAULT_SENDER" not in app.config:
             app.config["MAIL_DEFAULT_SENDER"] = default_settings.DEFAULT_MAIL_SENDER
 
+    if type(app.config["MAIL_DEFAULT_SENDER"]) == tuple:
+        warnings.warn(
+            "MAIL_DEFAULT_SENDER is not a tuple anymore and should be converted to a simple string."
```
Comment (almet) [inline ihatemoney/run.py:107]: We might want to provide the config line to change, since we have the tuple?

**Comment by almet** [comment]
Looks good to me, we can probably make the life of the admins easier by letting them know what the new config line should look like in their case.

**Comment by Glandos** [comment]
What? Helping admin? This is a weird requirement!
OK, I was looking for a Sunday joke. I implemented this, of course :+1:

File: ihatemoney/run.py [hunk 3]
```diff
@@ -103,13 +103,14 @@ def validate_configuration(app):
             app.config["MAIL_DEFAULT_SENDER"] = default_settings.DEFAULT_MAIL_SENDER
 
     if type(app.config["MAIL_DEFAULT_SENDER"]) == tuple:
+        (name, address) = app.config["MAIL_DEFAULT_SENDER"]
+        app.config["MAIL_DEFAULT_SENDER"] = f"{name} <{address}>"
         warnings.warn(
-            "MAIL_DEFAULT_SENDER is not a tuple anymore and should be converted to a simple string."
+            "MAIL_DEFAULT_SENDER has been changed from tuple to string."
+            + f" It was converted to '{app.config['MAIL_DEFAULT_SENDER']}'."
```
Comment (almet) [inline ihatemoney/run.py:110]: :+1:

**Pull request merged**